lib/services/concept_dictionary.dart

```diff
@@ -50,11 +54,36 @@ class ConceptDictionary {
     if (response.statusCode == 200) {
       var responseJson = jsonDecode(response.body);
       var resultList = responseJson['results'] ?? [];
+      var list = List<OmrsConcept>.from(resultList.map((v) => OmrsConcept.fromJson(v)));
       return List<OmrsConcept>.from(resultList.map((v) => OmrsConcept.fromJson(v)));
     } else {
       throw 'Failed to fetch Concept';
     }
   }
+  Future<List<OmrsConcept>> searchInvestigation(String term) async {
+    String? sessionId = await UserPreferences().getSessionId();
+    if (sessionId == null) {
+      throw 'Authentication Failure';
+    }
+    String url = '${AppUrls.omrs.concept}?q=$term&v=custom:(uuid,name,display,conceptClass:(uuid,name)';
+    var response = await http.get(
+      Uri.parse(url),
+      headers: <String, String>{
+        'Content-Type': 'application/json',
+        'Accept': 'application/json',
+        'Cookie': 'JSESSIONID=$sessionId',
+      },
+    );
+
+    if (response.statusCode == 200) {
+      var responseJson = jsonDecode(response.body);
+      var resultList = responseJson['results'] ?? [];
+      var list = List<OmrsConcept>.from(resultList.map((v) => OmrsConcept.fromJson(v)));
+      return list.where((element) => element.conceptClass?.name == 'Test'||element.conceptClass?.name == 'LabTest'||element.conceptClass?.name == 'Radiology'||element.conceptClass?.name == 'LabSet' || element.conceptClass?.name == 'Procedure').toList();
```

Comment:
Please do case insensitive match. Also, you may declare a string list with the classes that you want to match and then do a list.contains(element.conceptClass!.name)

Reply:
Done